### PR TITLE
Create dir /var/crash if it does not exist.

### PR DESCRIPTION
**Pull request by kwoodson**
```release-note
None
```

Saw an issue with rc.local on second boot fails because directory exists.
```
Aug 09 20:21:29 localhost.localdomain rc.local[1559]: + mkdir /var/crash
Aug 09 20:21:29 localhost.localdomain rc.local[1559]: mkdir: cannot create directory ‘/var/crash’: File exists
Aug 09 20:21:29 localhost.localdomain systemd[1]: rc-local.service: control process exited, code=exited status=1
Aug 09 20:21:30 localhost.localdomain systemd[1]: Failed to start /etc/rc.d/rc.local Compatibility.
Aug 09 20:21:30 localhost.localdomain systemd[1]: Unit rc-local.service entered failed state.
Aug 09 20:21:30 localhost.localdomain systemd[1]: rc-local.service failed.
```

### CODE REVIEW

File: pkg/vmimage/data/etc/rc.d/rc.local [hunk 1]
```diff
@@ -10,7 +10,7 @@ done
 
 # if there are multiple crashes, remove all but the most recent
 # mkdir is required or will fail rc.local
-mkdir /var/crash
+[[ -d /var/crash ]] || mkdir /var/crash
```
Comment (ehashman) [inline pkg/vmimage/data/etc/rc.d/rc.local]: Would this be simpler with `mkdir -p`?

Comment (kwoodson) [reply pkg/vmimage/data/etc/rc.d/rc.local]: I'm fine either way.

**Comment by kwoodson** [comment]
Updated to `mkdir -p`.  Let's go CI!

**Comment by ehashman** [comment]
Think that was a flake:

```
2019/08/12 13:56:41 Unsolicited response received on idle HTTP channel starting with \"HTTP/1.0 408 Request Time-out\\r\\nCache-Control: no-cache\\r\\nConnection: close\\r\\nContent-Type: text/html\\r\\n\\r\\n<html><body><h1>408 Request Time-out</h1>\\nYour browser didn't send a complete request in time.\\n</body></html>\\n\"; err=<nil>
2019/08/12 13:56:41 Unsolicited response received on idle HTTP channel starting with \"HTTP/1.0 408 Request Time-out\\r\\nCache-Control: no-cache\\r\\nConnection: close\\r\\nContent-Type: text/html\\r\\n\\r\\n<html><body><h1>408 Request Time-out</h1>\\nYour browser didn't send a complete request in time.\\n</body></html>\\n\"; 
```

**Comment by kwoodson** [comment]
Rerunning. 

I think the error was this:
```

570
[Fail] sync pod tests [EveryPR] [It] should not continuously update objects 
571
/home/prow/go/src/github.com/openshift/openshift-azure/test/e2e/specs/fakerp/sync.go:27
```

Which I think means that sync is rerunning and updating continually which might be this:
```
time="2019-08-12T14:16:06Z" level=info msg="Update SecurityContextConstraints.security.openshift.io/privileged"
time="2019-08-12T14:16:06Z" level=info msg="  unstructured.Unstructured{\n  \tObject: map[string]interface{}{\n  \t\t... // 6 identical entries\n  \t\t\"allowPrivilegedContainer\": bool(true),\n  \t\t\"allowedCapabilities\":      []interface{}{string(\"*\")},\n+ \t\t\"allowedUnsafeSysctls\":     []interface{}{string(\"*\")},\n  \t\t\"apiVersion\":               string(\"security.openshift.io/v1\"),\n  \t\t\"defaultAddCapabilities\":   nil,\n  \t\t... // 9 identical entries\n  \t\t\"seccompProfiles\":    []interface{}{string(\"*\")},\n  \t\t\"supplementalGroups\": map[string]interface{}{\"type\": string(\"RunAsAny\")},\n  \t\t\"users\": []interface{}{\n  \t\t\t... // 2 identical elements\n  \t\t\tstring(\"system:serviceaccount:openshift-etcd:etcd-backup\"),\n  \t\t\tstring(\"system:serviceaccount:openshift-azure-logging:geneva\"),\n- \t\t\tstring(\"system:serviceaccount:kube-system:sync\"),\n  \t\t},\n  \t\t\"volumes\": []interface{}{string(\"*\")},\n  \t},\n  }\n"
time="2019-08-12T14:16:07Z" level=debug msg="Waiting for the service catalog api to get aggregated"
time="2019-08-12T14:16:07Z" level=debug msg="Service catalog api is aggregated"
time="2019-08-12T14:16:07Z" level=debug msg="applying sc resources"
...
time="2019-08-12T14:19:10Z" level=info msg="sync done"
time="2019-08-12T14:22:04Z" level=info msg="starting sync"
...
time="2019-08-12T14:22:05Z" level=debug msg="applying other resources"
time="2019-08-12T14:22:07Z" level=info msg="Update SecurityContextConstraints.security.openshift.io/privileged"
time="2019-08-12T14:22:07Z" level=info msg="  unstructured.Unstructured{\n  \tObject: map[string]interface{}{\n  \t\t... // 6 identical entries\n  \t\t\"allowPrivilegedContainer\": bool(true),\n  \t\t\"allowedCapabilities\":      []interface{}{string(\"*\")},\n+ \t\t\"allowedUnsafeSysctls\":     []interface{}{string(\"*\")},\n  \t\t\"apiVersion\":               string(\"security.openshift.io/v1\"),\n  \t\t\"defaultAddCapabilities\":   nil,\n  \t\t... // 9 identical entries\n  \t\t\"seccompProfiles\":    []interface{}{string(\"*\")},\n  \t\t\"supplementalGroups\": map[string]interface{}{\"type\": string(\"RunAsAny\")},\n  \t\t\"users\": []interface{}{\n  \t\t\t... // 2 identical elements\n  \t\t\tstring(\"system:serviceaccount:openshift-etcd:etcd-backup\"),\n  \t\t\tstring(\"system:serviceaccount:openshift-azure-logging:geneva\"),\n- \t\t\tstring(\"system:serviceaccount:kube-system:sync\"),\n  \t\t},\n  \t\t\"volumes\": []interface{}{string(\"*\")},\n  \t},\n  }\n"
```
Something to do with this ^^

**Comment by ehashman** [comment]
```
time="2019-08-12T15:39:40Z" level=debug msg="400 Bad Request: Failed to apply request: UpdateWorkerAgentPoolDeleteVM: Code=\"InvalidResourceReference\" Message=\"Resource /subscriptions/225e02bc-43d0-43d1-a01a-17e584a4ef69/resourceGroups/CI-1842-E2E-CREATE-XCCIYJ/providers/Microsoft.Network/loadBalancers/KUBERNETES-INTERNAL referenced by resource /subscriptions/225e02bc-43d0-43d1-a01a-17e584a4ef69/resourceGroups/ci-1842-e2e-create-xcciyj/providers/Microsoft.Compute/virtualMachineScaleSets/ss-compute-1565622653 was not found. Please make sure that the referenced resource exists, and that both resources are in the same region.\"" func="github.com/openshift/openshift-azure/pkg/fakerp.(*Server).badRequest" file="/home/prow/go/src/github.com/openshift/openshift-azure/pkg/fakerp/util.go:24"
```

That's a new one...

**Comment by ehashman** [comment]
/retest

I think it's a flake.

**Comment by kwoodson** [comment]
Yay, currently flake free

@ehashman please do the honors.

**Comment by ehashman** [comment]
/retest

flake #1481 on e2e-create
flake #552 on upgrade-5.2

**Pull request merged**